src/NuGet.CommandLine/Commands/Command.cs

```diff
@@ -152,6 +156,36 @@ protected void SetDefaultCredentialProvider()
                 // add the proxy to v2 proxy cache.
                 v2ProxyCache?.Add(proxy);
             };
+            
+            NuGet.Protocol.Core.v3.HttpHandlerResourceV3.PromptForCredentials = uri =>
+            {
+                bool retrying = _credentialRequested.Contains(uri);
+
+                // Add uri to the hash set so that the next time we know the credentials for this
+                // uri has been requested before. In this case, NuGet will ask user for credentials.
+                if (!retrying)
+                {
+                    _credentialRequested.Add(uri);
+                }
+
+                return credentialProvider.GetCredentials(
+                    uri,
+                    proxy: null,
+                    credentialType: CredentialType.RequestCredentials,
+                    retrying: retrying);
+            };
+
+            var v2CredentialStoreType = typeof(NuGet.ICredentialCache).Assembly.GetType("NuGet.CredentialStore");
```

Comment:
We should not be doing reflection for this. Can you just make it public in NuGet.Core?